[Camera] Add Matter Camera TCs (TC-WEBRTC-1.1, TC-WEBRTC-1.2, TC-WEBRTC-1.3, TC-WEBRTC-1.4, TC-WEBRTC-1.5)

chulspro opened the pull request:
This PR Adds the following test cases:
- TC-WEBRTC-1.1
- TC-WEBRTC-1.2
- TC-WEBRTC-1.3
- TC-WEBRTC-1.4
- TC-WEBRTC-1.5

Controller support for 1.2, 1.3, 1.4, 1.5 will be submitted in the subsequent PRs

### Testing

Build python wheel and activate venv:

```
source ./scripts/activate.sh
./scripts/build_python.sh -i out/python_env
source out/python_env/bin/activate
```

To run tests for TC-WEBRTC-1.1:

```
./scripts/tests/run_python_test.py --factory-reset --app <chip-camera-app> --app-args "--trace-to json:log" --script src/python_testing/TC_WEBRTC_1_1.py --script-args "--commissioning-method on-network --qr-code MT:-24J0AFN00KA0648G00"
```

Commit:
Add TC_WEBRTC_1_1 to python_testing

Signed-off-by: abhishek.h <abhishek.h@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

Commit:
Add TC_WEBRTC_1_2 to python_testing

Signed-off-by: abhishek.h <abhishek.h@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

Commit:
Add TC_WEBRTC_1_3 to python_testing

Signed-off-by: abhishek.h <abhishek.h@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

Commit:
Add TC_WEBRTC_1_4 to python_testing

Signed-off-by: abhishek.h <abhishek.h@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

Commit:
Add TC_WEBRTC_1_5 to python_testing

Signed-off-by: abhishek.h <abhishek.h@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

restyled-commits committed:
Restyled by isort

Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

marktrayer commented on src/python_testing/TC_WEBRTC_1_3.py:
```diff
@@ -0,0 +1,153 @@
+            webrtc.SetCandidate(client, candidate)
+
+        self.step(7)
+        self.wait_for_user_input("Verify WebRTC session is established")
```
You should only do this if not running in ci; 
e.g. 
if not self.is_ci
 self.wait_for_user_input()

chulspro replied:
Thank you for your feedback. We fixed this issue with our PR

Commit:
ruff linting fix

Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

Commit:
Don't wait for user inetraction in CI - check peer connection state instead for session establishment

Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

yufengwangca commented on src/python_testing/TC_WEBRTC_1_1.py:
```diff
@@ -0,0 +1,168 @@
+from chip.webrtc import ErrorCallback_t, PeerConnectedCallback_t, PeerDisconnectedCallback_t, SdpAnswerCallback_t, StatsCallback_t
+from mobly import asserts
+from test_plan_support import commission_if_required
+
```
This test script isn't yet integrated with CI or TH.
Our standard method for CI integration uses the following meta session:

# === BEGIN CI TEST ARGUMENTS ===
# test-runner-runs:
#   run1:
#     app: ${CAMERA_APP}
#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
#     script-args: >
#       --PICS src/app/tests/suites/certification/ci-pics-values
#       --storage-path admin_storage.json
#       --commissioning-method on-network
#       --discriminator 1234
#       --passcode 20202021
#       --trace-to json:${TRACE_TEST_JSON}.json
#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
#     factory-reset: true
#     quiet: true
# === END CI TEST ARGUMENTS ===
#

CI will automatically detect and execute newly checked-in Python test case based on the info provided in the meta session. Do this integration in follow-up is ok.

Commit:
Exclude WEBRTC TCs from the CI.

Exclude WEBRTC TCs from the CI, until python webrtc bindings and
python controller changes are landed.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

Commit:
Update TCs based on controller refactor

Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

restyled-commits committed:
Restyled by prettier-yaml

Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

chulspro committed:
Merge branch 'master' into feature/contribution-testplans-webrtc

Commit:
Allocate video stream based on values from DUT

Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

chulspro commented on src/python_testing/TC_WEBRTC_1_1.py:
```diff
@@ -0,0 +1,145 @@
+        # This step is not from test plan
+        resolution = CameraAvStreamManagement.Structs.VideoResolutionStruct(width=640, height=480)
+
+        await self.send_single_cmd(
```
Thank you for your review. I updated it
